### ISSUE TITLE
New campaig hangs

### ISSUE THREAD

**Issue by iorq83** [MegaMek/mekhq#3701]
### Environment
0.49.13
Linux Mint 20.3
Java Version: 11.0.19

### Description

Generation Campaign hangs on Initializing New Campaign.

Last lines from log:
12:46:14,311 INFO  [mekhq.MekHQ] {main}
mekhq.MekHQ.initializeLogging(MekHQ.java:260) - Starting MekHQ v0.49.13
        Build Date: 2023-05-23T14:36:10.295482500
        Today: 2023-06-22
        Origin Project: MekHQ
        Java Vendor: Ubuntu
        Java Version: 11.0.19
        Platform: Linux 5.15.0-60-generic (amd64)
        System Locale: pl_PL
        Total memory available to MekHQ: 1 GB

12:46:19,541 INFO  [mekhq.campaign.finances.CurrencyManager] {SwingWorker-pool-2-thread-1}
mekhq.campaign.finances.CurrencyManager.registerCurrencies(CurrencyManager.java:192) - Starting load currency information from XML...
12:46:19,555 INFO  [mekhq.campaign.finances.CurrencyManager] {SwingWorker-pool-2-thread-1}
mekhq.campaign.finances.CurrencyManager.registerCurrencies(CurrencyManager.java:286) - Load of currency information complete!
12:46:19,621 INFO  [mekhq.campaign.personnel.ranks.Ranks] {SwingWorker-pool-2-thread-1}
mekhq.campaign.personnel.ranks.Ranks.initializeRankSystems(Ranks.java:113) - Starting Rank Systems XML load...
12:46:19,795 INFO  [mekhq.campaign.personnel.ranks.Ranks] {SwingWorker-pool-2-thread-1}
mekhq.campaign.personnel.ranks.Ranks.initializeRankSystems(Ranks.java:134) - Completed Rank System XML Load
12:46:19,982 INFO  [mekhq.campaign.universe.Factions] {SwingWorker-pool-2-thread-1}
mekhq.campaign.universe.Factions.loadDefault(Factions.java:146) - Starting load of faction data from XML...
12:46:20,016 INFO  [mekhq.campaign.universe.Factions] {SwingWorker-pool-2-thread-1}
mekhq.campaign.universe.Factions.loadDefault(Factions.java:150) - Loaded a total of 172 factions
12:46:20,086 INFO  [mekhq.campaign.personnel.Bloodname] {SwingWorker-pool-2-thread-1}
mekhq.campaign.personnel.Bloodname.loadBloodnameData(Bloodname.java:451) - Loaded 805 Bloodname records.
12:46:20,199 INFO  [mekhq.campaign.universe.Systems] {SwingWorker-pool-2-thread-1}
mekhq.campaign.universe.Systems.loadDefault(Systems.java:251) - Starting load of system data from XML...
12:46:24,126 INFO  [mekhq.campaign.universe.Systems] {SwingWorker-pool-2-thread-1}
mekhq.campaign.universe.Systems.loadDefault(Systems.java:256) - Loaded a total of 3445 systems in 3.926s.
12:46:24,160 WARN  [mekhq.campaign.universe.Systems] {SwingWorker-pool-2-thread-1}
mekhq.campaign.universe.Systems.logVeryCloseSystems(Systems.java:362) - Extremely close systems detected. Data error? Cambridge Perimeter Defense Station <-> Cambridge: 0.828 ly
12:46:24,164 WARN  [mekhq.campaign.universe.Systems] {SwingWorker-pool-2-thread-1}
mekhq.campaign.universe.Systems.logVeryCloseSystems(Systems.java:362) - Extremely close systems detected. Data error? Cambridge <-> Cambridge Perimeter Defense Station: 0.828 ly
12:46:24,488 INFO  [mekhq.gui.dialog.DataLoadingDialog.Task] {SwingWorker-pool-2-thread-1}
mekhq.gui.dialog.DataLoadingDialog$Task.doInBackground(DataLoadingDialog.java:260) - Starting a new campaign
12:46:28,144 INFO  [mekhq.campaign.CampaignOptions] {SwingWorker-pool-2-thread-1}
12:46:28,223 INFO  [mekhq.campaign.CampaignOptions] {SwingWorker-pool-2-thread-1}
mekhq.campaign.CampaignOptions.generateCampaignOptionsFromXml(CampaignOptions.java:3864) - Loading Campaign Options from Version 0.49.12-SNAPSHOT XML...
12:46:28,265 INFO  [mekhq.campaign.CampaignOptions] {SwingWorker-pool-2-thread-1}
mekhq.campaign.CampaignOptions.generateCampaignOptionsFromXml(CampaignOptions.java:3864) - Loading Campaign Options from Version 0.49.8-SNAPSHOT XML...
12:46:28,308 INFO  [mekhq.campaign.CampaignOptions] {SwingWorker-pool-2-thread-1}
mekhq.campaign.CampaignOptions.generateCampaignOptionsFromXml(CampaignOptions.java:3864) - Loading Campaign Options from Version 0.49.12-SNAPSHOT XML...
12:46:28,336 INFO  [mekhq.campaign.CampaignOptions] {SwingWorker-pool-2-thread-1}
mekhq.campaign.CampaignOptions.generateCampaignOptionsFromXml(CampaignOptions.java:3864) - Loading Campaign Options from Version 0.49.8-SNAPSHOT XML...
12:46:28,361 INFO  [mekhq.campaign.CampaignOptions] {SwingWorker-pool-2-thread-1}
mekhq.campaign.CampaignOptions.generateCampaignOptionsFromXml(CampaignOptions.java:3864) - Loading Campaign Options from Version 0.49.8-SNAPSHOT XML...
12:46:28,386 INFO  [mekhq.campaign.CampaignOptions] {SwingWorker-pool-2-thread-1}
mekhq.campaign.CampaignOptions.generateCampaignOptionsFromXml(CampaignOptions.java:3864) - Loading Campaign Options from Version 0.49.8-SNAPSHOT XML...

![mekhq](https://github.com/MegaMek/mekhq/assets/137400564/e513423a-01c8-4eda-bef1-4629f6ab702e)

**Comment by Windchild292** [comment]
Please increase your RAM allocation to 2GB. MHQ regularly runs out of memory below that point, and I'm going to add an error message on startup when <2GB is selected.

**Comment by iorq83** [comment]
Tried with more memory still the same. Changed java to temurin 11 and tried with 2GB too and still not going any further

**Comment by pheonixstorm** [comment]
Few additional questions. 
What is your total system memory and what was the maximum memory size that you tried?
Is this a clean install?
If clean, did you transfer any files over from an older version?
If not clean, did you unzip the latest version over a previous install?

**Comment by iorq83** [comment]
Total memory 16GB, I tried 8GB. Clean install without files from older version.

**Comment by pheonixstorm** [comment]
So I tried to start a new campaign and so far it works. The one thing that it may be is that the Select Campaign Preset box may be hiding behind the main HQ box, the one you did a screenshot of. The log file and progress bar (from the image) both match and it is the only thing I can think of at the moment. I would need to rebuild one of my older computers and install a recent copy of linux to test it but I am pretty sure this is the most likely cause. As for a solution (if this is the problem) I don't know. For that hopefully Windchild or another MM dev will have an answer.

**Comment by iorq83** [comment]
Hi, looks like You are right, window is in backgroud, I was able to resize it with alt+space menu but still cannot move it foreground :/

**Comment by pheonixstorm** [comment]
Well I hope the devs can come up with a solution for you. Did you try to do a search for open or closed issues like this? Seems odd that this would be the only time it has happened.

**Comment by Thom293** [comment]
alt-tab to the hidden window doesnt work?

**Comment by pheonixstorm** [comment]
Does Linux have an alt-tab function like Windows?

**Comment by iorq83** [comment]
> alt-tab to the hidden window doesnt work?

Yes linux has alt-tab function, but in start campaign I dont have proper window, so system cannot cycle thru it (I can see other windows but not this one). After mekhq starts there is window I can alt-tab to but no in time of initializing new campaign. Looks like I managed to alt-space move window from background of this nice picture after carefully find a pice of menu, but pain in the "some place" :)

**Comment by pheonixstorm** [comment]
What screen resolution are you using?
Are you able to hit enter to move it to the next part of the campaign setup screen?

**Comment by iorq83** [comment]
Sorry for slow response
dual screen:
first 1600x900
second 1920x1080
but mekhq window appears on smaller one.
Not reacting on enter hit.

**Comment by SGT-Maxwell** [comment]
I get the exact same issue on Arch Linux

**Comment by pachorrus** [comment]
Hi everybody,

I have the same problem (i am running ubuntu). But i can move the type-of-campaign window from behind the image. Just using alt-space when the mouse pointer is around the center of the image, the pop-up window appears, and i click on Move instead of Resize. Then, the pointer becomes a kind of a cross. Now i move the mouse (without clicking) out of the image and the type-of-campaign window appears from behind the image when the mouse pointer is outside of the image. Once the window is outside, you can whatever is needed.

Thank you very much for the tip, because i would not have been able of figuring it out.

**Comment by IllianiCBT** [comment]
Can we have confirmation whether this is still an ongoing issue?

**Comment by firefly2442** [comment]
I'm seeing the same thing with PopOS (variant of Ubuntu) with 781610b5a2c5191f10e0d76041fba979923e78f9.

**Comment by IllianiCBT** [comment]
@firefly2442 to confirm, this still happens with CO IIC?

**Comment by firefly2442** [comment]
> [@firefly2442](https://github.com/firefly2442) to confirm, this still happens with CO IIC?

CO IIC?  I haven't seen any issues since merging it in on Linux.
